### DISCUSSION TITLE
Future of Deadname Remover

### DISCUSSION THREAD

**Discussion by WillowHayward** [arimgibson/Deadname-Remover#442] [Announcements]
Hi everyone,

First of all I'd like to apologize for how absent I've been regarding this project. I started my first full time dev job about a year ago, and also came out as trans (GEE WHAT A TWIST) so it's been a pretty wild time. Still, I regret not being more active in here and I'd like to apologize for that.

My top priority right now is to try and distribute the ability to upload new versions to Chrome/Firefox/whatever so that we don't have to rely on my individual availability to keep things up to date.

As well as that, several school districts have banned the extension. I'm currently in contact with a few of them to try and determine a solution for this.

Beyond that, I want to expand the functionality to allow for multiple names, better handling of non-ASCII characters, and better support with Google's services. 

If anyone has any questions, suggestions, or other feel free to make a new discussion or comment on here.

**Comment by arimgibson** [comment]
Haha I love it, ngl I always figured you were trans-masc but am so glad to hear of the coming out and hope it's going good! Not meaning to spite anyway, I've been tinkering a bit on the side with a fork of this similar concept as I want to be available to consistently maintain something like this. I also felt like some of the code in this repo seems a bit overkill, so I wanted to try and simplify a bit.

I mention that as I want to help keep this concept going in whatever way makes the most sense, whether that be a diff repo or taking a bigger role in maintaining this one. I'm not sure the best way to do that and am definitely feeling biased towards what I've got going right now in a diff repo, although it's still a work in progress. I've also been trying to work in some of the suggestions this repo has received in the first release (if I end up releasing it haha). Would be curious if you have any thoughts on that and plans going forward.

Also want to mention that at least here in the US, I know my previous school district and districts attended by friends would whitelist extensions rather than blacklisting the ones they didn't like.

**Reply by WillowHayward** [reply]
I am in complete and total support of having multiple tools for the same purpose floating around! The first version of this was a total hack to get the job done that I barely thought about until it started to get some traction with covid. It originally just did a find a replace a few times a second. CPU _loved_ that, lemme tell you. Another user fixed that nonsense.

The biggest advantage of "mine" right now is that it has an established audience (and frankly a pretty solid name), and that's something I think should be shared - I reckon if we lightly coordinate and share resources that'll be the best option for everyone. And yeah, some of the code definitely needs a bit of a simplification haha.

If you are interested in becoming a maintainer of this one (independent of whether you continue with your project) I'm all for it - lemme know!

**Reply by WillowHayward** [reply]
Also I'm cracking up at being confused for trans-masc, just because I don't think that's ever happened to me before. Irl I'm 6 feet tall, shoulders as broad as they come, and usually with a thick layer of stubble so the thought of being viewed as anything other than a cis man is still a bit foreign to me.

**Reply by WillowHayward** [reply]
(Not to suggest trans-masc individuals don't/can't have any of those properties, of course)

**Reply by arimgibson** [reply]
I'm so sorry for the late reply!! I have been meaning to get back to you; it's been a busy couple weeks 😛 

Lmao I love that with the find and replace... I can definitely say I'm guilty of janky solutions like that too haha. I will definitely let you know if I want to be a maintainer; right now my schedule is crazy and I'm still trying to work out some tweaks on my version. I do appreciate how well established this extension is though, and hate to reinvent the wheel. When I've got something good enough going on, I'll send it over if you want to take a look!

Haha I guess I had just assumed that you must already be trans if you're working on this project, but I do that a lot. I started out as the "ally" working for queer rights in schools before it all hit me. _one of us, one of us, one of us!_
